class_2.tex

```diff
@@ -106,55 +119,88 @@ \subsection{Lattices and Fixpoints}
+
+\begin{definition}[Lattice]
+    A poset $(A,\sqsubseteq)$ is a \emph{lattice} if for all $x$, $y$ in $A$, $\bigsqcup \{x,y\}$ ($x \sqcup y$, join) and $\bigsqcap \{x,y\}$ ($x \sqcap y$, meet) exist.
+\end{definition}
 
 \begin{definition}[Complete Lattice]
-    Poset $(A,\sqsubseteq)$ is a \emph{complete-lattice} if for all $B \subseteq A$, both $\bigsqcap B$ and $\bigsqcup B$ exist.
+    A poset $(A,\sqsubseteq)$ is a \emph{complete lattice} if for all $B \subseteq A$, $\bigsqcap B$ and $\bigsqcup B$ exist.
 \end{definition}
+
 \begin{example}
-    Let $(A,\sqsubseteq)$ be a complete-lattice. Then: $\bigsqcup A = \top$, $\bigsqcap A = \bot$, $\bigsqcup \varnothing = \bot$, and $\bigsqcap \varnothing = \top$
+    Let $(A,\sqsubseteq)$ be a complete-lattice. Then: $\bigsqcup A = \top$, $\bigsqcap A = \bot$, $\bigsqcup \varnothing = \bot$, and $\bigsqcap \varnothing = \top$. $\top$ and $\bot$ are called top and bottom, respectively.
 \end{example}
 
-% \begin{definition}
-%     Let $(A,\sqsubseteq)$ be a poset and let $F$ be a function from $A$ to $A$. 
-%     Then $x$ in $A$ is a pre-fixpoint of $F$ if $x \sqsubseteq F(x)$ and is a post-fixpoint of $F$, if $F(x) \sqsubseteq x$. 
-% \end{definition}
-
+\begin{definition}[Monotone Function]
+Let $(A, \sqsubseteq)$ be a poset. A function $F: A \to A$ is \emph{monotone} if for all $x$, $y$ in $A$, if $x \sqsubseteq y$ then $F(x) \sqsubseteq F(y)$.
+\end{definition}
 
+\begin{definition}
+Let $(A, \sqsubseteq)$ be a poset and $F: A \to A$. An element $x$ in $A$ is,
+\begin{enumerate}
+  \item a \emph{prefixpoint} of $F$ if $x \sqsubseteq F(x)$,
+  \item a \emph{postfixpoint} of $F$ if $F(x) \sqsubseteq x$.
+\end{enumerate}
+\end{definition}
 
+% TODO
+\begin{definition}[Fixpoint]
+Let $(A, \sqsubseteq)$ be a poset and $F: A \to A$. An element $x$ in $A$ is a \emph{fixpoint} of $F$ if it is both a \emph{pre-} and \emph{postfixpoint} of $F$.
+\end{definition}
 
 \begin{theorem}[Knaster-Tarski]
 \label{thrm:knaster}
-    For every complete lattice $(A,\sqsubseteq)$ and monotone function $f$ on $A$, it holds that: (i) $\bigsqcup \{x\in A| x \sqsubseteq f(x)\}$ is the unique greatest fixpoint of $F$; (ii) $\bigsqcap \{x \in A| f(x) \sqsubseteq x\}$ is the unique least fixpoint of $f$. 
+    Let $(A,\sqsubseteq)$ be a complete lattice and let $F$ a monotone function on $A$.
+    Further, let $\hat{y} = \{x \mid x \sqsubseteq F(x)\}$ and $\hat{z} = \{x \mid F(x) \sqsubseteq x\}$.
+    It holds, that:
+    \begin{enumerate}
+      \item $\hat{y}$ and $\hat{z}$ are fixpoints of $F$,
+      \item for all fixpoints $x$ of $F$, $\hat{y} \sqsubseteq x \sqsubseteq \hat{z}$
+    \end{enumerate}
 \end{theorem}
 
+{\flushleft We write $\lfp(F) = \{x \mid x \sqsubseteq F(x)\}$ and $\gfp(F) = \{x \mid F(x)\sqsubseteq x\}$, to denote the least fixpoint and the greatest fixpoint, respectively.}
```

Comment:
here the greatest fixpoint should be the set of the least fixpoint and viceversa. Also you are missing to take the least upper bound and the greatest lower bounds.

Reply:
Thanks... That was stupid